src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.Node.cs

```diff
@@ -72,5 +78,87 @@ private string GetDebuggerDisplay()
                 return NameSpan + ", " + ParentIndex;
             }
         }
+
+        public readonly struct ParameterTypeInfo
+        {
+            public string Name { get; }
+            public bool IsComplexType { get; }
+
+            public ParameterTypeInfo(string name, bool isComplex)
+            {
+                Name = name;
+                IsComplexType = isComplex;
+            }
+
+            public override string ToString()
+                => $"{Name}, {IsComplexType}";
+        }
+
+        public readonly struct ExtensionMethodInfo
+        {
+            public string FullyQualifiedContainerName { get; }
+
+            public string Name { get; }
+
+            public ExtensionMethodInfo(string fullyQualifiedContainerName, string name)
+            {
+                FullyQualifiedContainerName = fullyQualifiedContainerName;
+                Name = name;
+            }
+        }
+
+        private sealed class ParameterTypeInfoProvider : ISignatureTypeProvider<ParameterTypeInfo, object>
+        {
+            public static readonly ParameterTypeInfoProvider Instance = new ParameterTypeInfoProvider();
+
+            private static ParameterTypeInfo ComplexInfo
+                => new ParameterTypeInfo(string.Empty, isComplex: true);
+
+            public ParameterTypeInfo GetPrimitiveType(PrimitiveTypeCode typeCode)
+                => new ParameterTypeInfo(typeCode.ToString(), isComplex: false);
+
+            public ParameterTypeInfo GetArrayType(ParameterTypeInfo elementType, ArrayShape shape) => ComplexInfo;
+
+            public ParameterTypeInfo GetSZArrayType(ParameterTypeInfo elementType) => ComplexInfo;
+
+            public ParameterTypeInfo GetGenericInstantiation(ParameterTypeInfo genericType, ImmutableArray<ParameterTypeInfo> typeArguments)
```

Comment:
pity this will still allocate.  can you check with @tmat if there's a way to avoid that?

Reply:
See comment here:
https://github.com/dotnet/roslyn/pull/39126#discussion_r339848845

Reply:
#resolved